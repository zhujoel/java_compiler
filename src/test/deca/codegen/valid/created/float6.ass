; start main program
; Class Declaration
; Code de la table des méthodes de Object
	LOAD #null, R0
	STORE R0, 1(GB)
	LOAD code.Object.equals, R0
	STORE R0, 5(GB)
; Main Function
; Test pour savoir si la pile est pleine
	TSTO #4
	BOV pile_pleine
	ADDSP #4
; Declaration des variables
	LOAD #0x1.266666p1, R2
	STORE R2, 12(GB)
	LOAD #0x1.333334p0, R2
	STORE R2, 13(GB)
; Beginning of main instructions:
; *
	LOAD #0x1.333334p0, R2
	LOAD #0x1.4p1, R3
	MUL R2, R3
	BOV debordement
	STORE R3, 12(GB)
	LOAD 12(GB), R1
	WFLOAT
	HALT
pile_pleine:
	WSTR "Erreur : Pile pleine."
	WNL
	ERROR
dereferencement_null:
	WSTR "Erreur : dereferencement null."
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : Tas plein."
	WNL
	ERROR
debordement:
	WSTR "Erreur : debordement durant une operation arithmétique"
	WNL
	ERROR
io_error:
	WSTR "Erreur : input ou output"
	WNL
	ERROR
; end main program
