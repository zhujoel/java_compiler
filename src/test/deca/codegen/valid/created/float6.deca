{
    float a = 2.3;
    float b = 1.2;
    a = 1.2 * 2.5;
    print(a);
}
