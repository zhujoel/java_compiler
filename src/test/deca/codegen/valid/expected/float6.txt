3.00000e+00
